### ISSUE TITLE
BUG: same feature column creates duplicate tensors for DNNLinearCombinedRegressor

### ISSUE THREAD

**Issue by facaiy** [tensorflow/tensorflow#13175]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Mac OS X 10.11.6
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.3.0
- **Python version**: 3.5.2
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

### Describe the problem

It seems that the same feature column creates two tensors for `DNNLinearCombinedRegressor` (see `voc_embed` in graph below). Is this the right behavior we expected?

The behavior stems from that feature column is processed by DNN and Linear independently, see code: [dnn](https://github.com/tensorflow/tensorflow/blob/b46340f40fe5e2ec9bfcd385b07cfb914055fb51/tensorflow/python/estimator/canned/dnn_linear_combined.py#L154) and [linear](https://github.com/tensorflow/tensorflow/blob/b46340f40fe5e2ec9bfcd385b07cfb914055fb51/tensorflow/python/estimator/canned/dnn_linear_combined.py#L196)

### Source code / logs

```python
# coding: utf-8
import tensorflow as tf
from tensorflow import feature_column as fc

f = fc.embedding_column(
        fc.categorical_column_with_vocabulary_list("voc", ["a", "b"]),
        dimension=8)
e = tf.estimator.DNNLinearCombinedRegressor(
        model_dir="/tmp/tf/test_dnn",
        linear_feature_columns=[f],
        dnn_feature_columns=[f],
        dnn_hidden_units=[10])

def input_fn():
    return ({"voc": tf.identity(
                        tf.constant(["a", "b", "a"], name="voc_input"),
                        name="identity")},
            tf.constant([1, 0, 1]))

e.train(input_fn, steps=10)
```

![graph-large_attrs_key _too_large_attrs limit_attr_size 1024 run 1](https://user-images.githubusercontent.com/1112263/30632078-63853988-9e19-11e7-94c8-c660c6ed55e2.png)

**Comment by poxvoculi** [comment]
It doesn't look like ideal behavior, but perhaps it's not unexpected.  @ispirmustafa

**Comment by ispirmustafa** [comment]
yes. it's not ideal. we can improve it.

**Comment by facaiy** [comment]
How about converting `feature_column` to `_LazyBuilder` at first, and the feeding `_LazyBuilder` to DNN and LR? I guess the solution could solve the problem (and is lightweight).

**Comment by facaiy** [comment]
Oh, I forget the optimizer problem: if a tensor is owned by both DNN and LR network, does it bring conflict as the tensor is trained by two optimizer?

**Comment by facaiy** [comment]
Hi, @ispirmustafa . Does anyone work on it? Do you have any good idea? Thanks.

**Comment by ispirmustafa** [comment]
Hi @facaiy ,
as you suggested using _LazyBuilder works. We should not open _LazyBuilder as a public API though. 
also sending _LazyBuilder around will complicate the underlying functions. 
If you find a way which does not complicate API, would be happy to review it.
nobody is working on it yet. contributions welcome.

**Comment by facaiy** [comment]
Sure, I'd like to take a try. Thank you for explanation, @ispirmustafa .

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by facaiy** [comment]
Sorry for the delay. I found the problem is in a dilemma, because feature columns use different optimizer for DNN and Linear part.

I'll close the issue since I have no good idea of how to fix it, and duplication features are excepted for me.
